crates/sargon/src/hierarchical_deterministic/bip32/key_space/traits/is_in_local_key_space.rs

```diff
@@ -1,6 +1,27 @@
-use crate::{KeySpace, U31};
+use crate::prelude::*;
 
-pub trait IsInLocalKeySpace {
+pub trait IsMaybeKeySpaceAware {
+    fn maybe_key_space(&self) -> Option<KeySpace>;
+}
+
+pub trait IsKeySpaceAware: IsSecurityStateAware {
     fn key_space(&self) -> KeySpace;
+}
+
+impl<T: IsKeySpaceAware> IsMaybeKeySpaceAware for T {
+    fn maybe_key_space(&self) -> Option<KeySpace> {
+        Some(self.key_space())
+    }
+}
+impl<T: IsKeySpaceAware> IsSecurityStateAware for T {
+    fn is_securified(&self) -> bool {
+        self.key_space().is_securified()
+    }
+}
+
+pub trait HasIndexInLocalKeySpace {
     fn index_in_local_key_space(&self) -> U31;
 }
+pub trait IsInLocalKeySpace: HasIndexInLocalKeySpace + IsKeySpaceAware {}
+
+impl<T: HasIndexInLocalKeySpace + IsKeySpaceAware> IsInLocalKeySpace for T {}
```

Comment:
This is essentially a "shortcut"/""typealias"" of traits :) it allows us to constraint some generic `U: IsInLocalKeySpace` instead of having to write `U: HasIndexInLocalKeySpace + IsKeySpaceAware`. But since it is only a "shortcut" we do want to consider all types which are `HasIndexInLocalKeySpace + IsKeySpaceAware` as in fact `IsInLocalKeySpace`. So:

`impl<T: HasIndexInLocalKeySpace + IsKeySpaceAware> IsInLocalKeySpace for T {}` gives exactly that! Rust magic! 🦀